docs/rfcs/038-independent-compute-release.md

```diff
@@ -0,0 +1,298 @@
+
+Now, we practice a manual v0 independent compute release flow -- after getting a new compute release
+image and tag, we pin it region by region using Admin UI. It's better, but still has its own flaws:
+
+1. It's simple, but fairly manual process, as you need to click through a few pages.
+2. It's prone to human errors, e.g. you could mistype or copy the wrong compute tag.
+3. We now require an additional approval in the Admin UI, which partially solves the 2.,
+   but also makes the whole process pretty annoying, as you constantly need to go back
+   and forth between two people.
+
+## Non goals
+
+It's not a goal of this document to propose a design of some general purpose release tool like Helm.
+The document considers how the current compute fleet is orchestrated at Neon. Even if we later
+decide to split control plane further (e.g. introduce a separate compute controller), the proposed
+release process shouldn't change much, i.e. the releases table and API will reside in
+one of the parts.
+
+Achieving the canary and/or blue-green deploy strategies is out of the scope of this document. They
+were kept in mind, though, so it's expected that the proposed approach will lay down the foundation
+for implementing them in future iterations.
+
+## Impacted components
+
+Compute, control plane, CI.
+
+## Prior art
+
+One of the very close examples is how Helm tracks [releases history](https://helm.sh/docs/helm/helm_history/).
+
+In the code:
+
+- [Release](https://github.com/helm/helm/blob/2b30cf4b61d587d3f7594102bb202b787b9918db/pkg/release/release.go#L20-L43)
+- [Release info](https://github.com/helm/helm/blob/2b30cf4b61d587d3f7594102bb202b787b9918db/pkg/release/info.go#L24-L40)
+- [Release status](https://github.com/helm/helm/blob/2b30cf4b61d587d3f7594102bb202b787b9918db/pkg/release/status.go#L18-L42)
+
+TL;DR it has several important attributes:
+
+- Revision -- unique release ID/primary key. Not the same as application version,
+  because the same version can be deployed several times, e.g. after newer version rollback.
+- App version -- version of the application chart / code.
+- Config -- set of overrides to the default config of the application.
+- Status -- current status of the release in the history.
+- Timestamps -- tracks when release was created and deployed.
+
+## Proposed implementation
+
+### Separate release branch
+
+We will use a separate release branch `release-compute` to have a clean history fo releases and commits.
+In order to avoid confusion with storage releases, we will use a different prefix for compute [git release
+tags](https://github.com/neondatabase/neon/releases) -- `release-compute-XXXX`. It probably still makes
+sense to use just `release-XXXX` for docker images, because `neondatabase/compute-node-vXX` is a separate
+Docker image from `neondatabase/neon` and `neondatabase/compute-node-v16:release-compute-XXXX` will just
+look awkward. Yet, it doesn't matter much, so we could do either of two.
```

Comment:
I'm for using `neondatabase/compute-node-v16:release-compute-XXXX` — it looks weird, but unambiguous and matches with git tags, so less chance that we confuse storage with compute release by having `release-XXXX`

Reply:
Yeah, makes sense, changed